src/getTableauServerVersion.ts

```diff
@@ -0,0 +1,37 @@
+import { ZodiosError } from '@zodios/core';
+import { fromError, isZodErrorLike } from 'zod-validation-error';
+
+import { getConfig } from './config.js';
+import { RestApi } from './sdks/tableau/restApi.js';
+import { ProductVersion } from './sdks/tableau/types/serverInfo.js';
+import { ExpiringMap } from './utils/expiringMap.js';
+import { getExceptionMessage } from './utils/getExceptionMessage.js';
+
+let tableauServerVersions: ExpiringMap<string, ProductVersion> | undefined;
+
+export const getTableauServerVersion = async (server: string): Promise<ProductVersion> => {
+  if (!tableauServerVersions) {
+    tableauServerVersions = new ExpiringMap<string, ProductVersion>({
+      expirationTimeMs: getConfig().tableauServerVersionCheckIntervalInHours * 60 * 60 * 1000,
+    });
+  }
+
+  const serverVersion = tableauServerVersions.get(server);
+  if (serverVersion) {
+    return serverVersion;
+  }
```

Comment:
We cache the version based off the server host for 1 hour (by default). The server host is currently static (based off the `SERVER` environment variable) but once OAuth support lands the MCP server could be pointed at Tableau Cloud and the user could sign into *any* pod. This approach should work for both scenarios.